metadata/metadata_test.go

```diff
@@ -555,6 +555,56 @@ var _ = Describe("Metadata", func() {
 		})
 	})
 
+	When("SetAnnotation is called", func() {
+		It("should add the annotation to the object", func() {
+			key := "foo"
+			value := "bar"
+			pod := &corev1.Pod{}
+
+			Expect(pod.Annotations).To(HaveLen(0))
+			Expect(SetAnnotation(pod, key, value)).To(Succeed())
+			Expect(pod.Annotations).To(HaveKeyWithValue(key, value))
+		})
+
+		It("should add the annotation to the existing ones", func() {
+			key := "foo"
+			value := "bar"
+			pod := &corev1.Pod{
+				ObjectMeta: v1.ObjectMeta{
+					Annotations: map[string]string{"quux": "corge"},
+				},
+			}
+
+			Expect(pod.Annotations).To(HaveLen(1))
+			Expect(SetAnnotation(pod, key, value)).To(Succeed())
+			Expect(pod.Annotations).To(HaveLen(2))
+			Expect(pod.Annotations).To(HaveKeyWithValue(key, value))
```

Comment:
Not checking here if the existing ones changed. Use `MatchAllKeys` (check other tests in this very same file).